components/fxa-client/src/device.rs

```diff
@@ -96,14 +98,16 @@ impl FirefoxAccount {
     }
 
     pub(crate) fn invoke_command(
-        &self,
+        &mut self,
```

Comment:
So just to check that I understand...all this mutable-self is because these methods might now alter the persisted account state by completing the migration, right?

Reply:
I recall a bug a while ago where Fenix was deserializing multiple copies of the `FirefoxAccount` object from JSON, one in the main app and then others in background worker threads. Do we still have any risk of that here? If so then I wonder if we might have some strange behaviours lurking where we try to complete the migration multiple times, possibly causing the FxA server's device-registration logic to get confused.

/cc @grigoryk

Reply:
> So just to check that I understand...all this mutable-self is because these methods might now alter the persisted account state by completing the migration, right?

Yes it makes me nervous too. I almost wonder if we should go with a different approach:
- return boolean on `fn migrate`
- let a-c call the same method later if we returned false.